services/telemetry/telemetry.go

```diff
@@ -267,6 +268,8 @@ func (s *Service) makeV2Payload(serverUUID string) (*reporter.ReportRequest, err
 		Version:            s.pmmVersion,
 		UpDuration:         ptypes.DurationProto(time.Since(s.start)),
 		DistributionMethod: s.tDistributionMethod,
+		SttEnabled:         wrapperspb.Bool(settings.SaaS.STTEnabled),
```

Comment:
with this approach, it never be nil. Do we need still wrapper around it? Just asking, because I don't know

Reply:
this new code won't send nils, but the older PMM versions won't send the two field, so when on telemetryd we unmarshall an event received from older pmm servers which don't send these two fields, they will be filled as nil and we can perform nil checks in that case https://github.com/percona-platform/telemetryd/pull/161/files#diff-a9ef45466a6e12ccc15711a5dd88db525d8928020c2e093f5e467fc7e7ffe436R300-R308  . at least that was my idea behind it. correct me if I am wrong or mixing things up

Reply:
ok so this wrapperspb has a weird behaviour, it doesn't set the value if it is `false`  `wrapperspb.Bool(false)`, only sets the value for true  https://github.com/golang/protobuf/issues/438

weird

Reply:
will have to do this to be able to set false

```
if settings.SaaS.STTEnabled {
		event.SttEnabled = &wrapperspb.BoolValue{Value: true}
	} else {
		event.SttEnabled = &wrapperspb.BoolValue{Value: false}
	}

	if settings.IntegratedAlerting.Enabled {
		event.IaEnabled = &wrapperspb.BoolValue{Value: true}
	} else {
		event.IaEnabled = &wrapperspb.BoolValue{Value: false}
	}
```

then it does set false, but if I try to print event with a field set to false it still wont show `stt_enabled:{}`

i hate this wrapper

Reply:
It won't help, `&wrapperspb.BoolValue{Value: false}` == `wrapperspb.Bool(false)`

Problem not with the function, but with type
```
type BoolValue struct {
	state         protoimpl.MessageState
	sizeCache     protoimpl.SizeCache
	unknownFields protoimpl.UnknownFields

	// The bool value.
	Value bool `protobuf:"varint,1,opt,name=value,proto3" json:"value,omitempty"`
}
```

Field value has `omitempty` annotation, and for the `bool` fails is equal to emtpy.

Reply:
I suppose you need some enum `ENABLED, DISABLED, UNKNOWN` instead of bool here.

Reply:
>  it doesn't set the value if it is false

That's ok, in protobuf bool `false` == `empty`, and that's because you used `wrapperspb`, to have a pointer to bool wrapper instead of plain `bool`.

Reply:
ok I will remove this, and test it once on the dev deployment

Reply:
works